### PR TITLE
Rework website home & download pages

### PR DESCRIPTION
**Pull request by starkos**
**What does this PR do?**

Reworks the placeholder home and download pages on the new Docusaurus-based website.

**How does this PR change Premake's behavior?**

No changes to Premake itself.

**Anything else we should know?**

I still need to rework the header and footer, and move some content from `docs/` to `pages/` such as the help and "who uses" pages. Tracking the work on #1547.

**Did you check all the boxes?**

- [x] Focus on a single fix or feature; remove any unrelated formatting or code changes
- [ ] ~Add unit tests showing fix or feature works; all tests pass~
- [x] Mention any [related issues](https://github.com/premake/premake-core/issues) (put `closes #XXXX` in comment to auto-close issue when PR is merged)
- [x] Follow our [coding conventions](https://github.com/premake/premake-core/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] Minimize the number of commits

*You can now [support Premake on our OpenCollective](https://opencollective.com/premake). Your contributions help us spend more time responding to requests like these!*

### CODE REVIEW

File: website/src/pages/download.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+			<Container>
+				<Row>
+					<Column>
+						<h3>Pre-Built Binaries</h3>
+						<p>Binaries simply need to be unpacked and placed somewhere on the system search path or any other convenient location.</p>
+						<p>
+							<DownloadLink arch="windows" /><br />
+							<DownloadLink arch="linux" /><br />
+							<DownloadLink arch="macos" /><br />
+						</p>
```
Comment (KyrietS) [inline website/src/pages/download.js:68]: `<div>` cannot appear as a descendant of `<p>`. You will see this error when you go to download page and open console in your browser. It's not react error. Website is compiled correctly but accorting to W3C you cannot put *block elements* inside `<p>` and React is kind enough to report this in the console.

My suggestion here is to change this paragraph with `DownloadLink`s into list `<ul>`, `<li>` then add a class to `<ul>`, e.g. "download-links" and style it. (then you can remove "download-link" class from `DownloadLink`). Unlike `<p>`, lists can contain *divs* and other *block elements*.

* To remove list bullets use: `list-style: none;`
* You can also style each item in the list: `.download-links > li { /*...*/ }`

File: website/src/pages/download.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+					<Column>
+						<h3>Build It Yourself</h3>
+						<p>
+							The source code package includes project files for all supported toolsets.
+							See <Link to="https://github.com/premake/premake-core/blob/master/BUILD.txt">BUILD.txt</Link> for
+							build instructions.
+						</p>
+						<p>
+							<DownloadLink arch="src" />
+						</p>
```
Comment (KyrietS) [inline website/src/pages/download.js:79]: Same problem as above. I suggest using list with one item. Left padding is natural in lists. You don't have to create it artificially for every `DownloadLink`.

File: website/src/pages/download.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+		filename = 'windows.zip';
+		label = 'Windows';
+		break;
+	}
+	return (
+		<div className="download-link">
+			<Link to={`https://github.com/premake/premake-core/releases/download/v${LATEST_VERSION}/premake-${LATEST_VERSION}-${filename}`}>
+				<b>{label}</b>
+			</Link>
+		</div>
```
Comment (KyrietS) [inline website/src/pages/download.js:37]: I don't think that using `div` here and styling it in this way is a good idea. I suggest removing `div` so the  `Link` becomes the top element. To understand my point of view please look at my comments below.

File: website/src/pages/index.js [hunk 4]
```diff
@@ -1,104 +1,79 @@
...
+			</Link>
+		</div>
+	</Container>;
+
+
+const Feature = ({ title, children }) =>
+	<div className="col col--4">
+		<h3>{title}</h3>
+		<div>{children}</div>
+	</div>;
```
Comment (KyrietS) [inline website/src/pages/index.js:33]: I suggest using `<Column> </Column>` here. It's already imported but it's not used (yet).

**Review by KyrietS — CHANGES_REQUESTED** [review]
> I still need to rework the header and footer, and move some content from docs/ to pages/ such as the help and "who uses" pages.

Just a reminder that you can create simple pages in Markdown. Docusaurus will take care of it. You don't have to use JS and React in every page.

**Comment by starkos** [comment]
Good feedback, thanks! I was going back and forth on the layout of the download links and didn't clean things up properly; I've switched them to `<ul>` as suggested. I'm hoping/intending to loop back and associate icons with each one at some point (why I wrapped them in `<div>`).

Console errors are gone. I also backed out my formatting change to `docusaurus.config.js` to avoid conflicts with your commit.

**Review by KyrietS — APPROVED** [review]
Thanks for the changes you've made. It looks great now. Those `DownloadLink` lists still can be simplified and refactored but it's good enough to merge this commit in my opinion.

And I wouldn't worry about conflicts in `docusaurus.config.js`. I can always do rebase and force-push to my branch :-)

**Pull request merged**